addons/website_sale/controllers/main.py

```diff
@@ -485,13 +485,15 @@ def checkout_form_validate(self, mode, all_form_values, data):
         # vat validation
         Partner = request.env['res.partner']
         if data.get("vat") and hasattr(Partner, "check_vat"):
-            check_func = request.website.company_id.vat_check_vies and Partner.vies_vat_check or Partner.simple_vat_check
-            vat_country, vat_number = Partner._split_vat(data.get("vat"))
-            if not check_func(vat_country, vat_number):
-                if country.code and check_func(country.code, data["vat"]):
```

Comment:
Oh right, I did not know the check_vat did that (it was done recently). So yes it makes sense to harmonize.